RiotSwiftUI/Modules/Room/UserSuggestion/Service/UserSuggestionService.swift

```diff
@@ -48,22 +48,61 @@ class UserSuggestionService: UserSuggestionServiceProtocol {
     private let roomMembersProvider: RoomMembersProviderProtocol
     
     private var suggestionItems: [UserSuggestionItemProtocol] = []
+    private let currentTextTriggerSubject = CurrentValueSubject<String?, Never>(nil)
+    private var cancellables = Set<AnyCancellable>()
     
     // MARK: Public
     
-    var items: CurrentValueSubject<[UserSuggestionItemProtocol], Never>
-    var currentTextTrigger: String?
+    var items = CurrentValueSubject<[UserSuggestionItemProtocol], Never>([])
+    
+    var currentTextTrigger: String? {
+        currentTextTriggerSubject.value
+    }
     
     // MARK: - Setup
     
     init(roomMembersProvider: RoomMembersProviderProtocol) {
         self.roomMembersProvider = roomMembersProvider
-        self.items = CurrentValueSubject([])
+        
+        currentTextTriggerSubject
+            .debounce(for: 0.5, scheduler: RunLoop.main)
```

Comment:
You could also add a `.removeDuplicates()` after the debounce to avoid extra fetches if someone types a character and then deletes it.

Reply:
Good idea! 👍